Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -185,6 +186,77 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::starts_with)
     return Value(string.substring(start, search_string_length) == search_string);
 }
 
+static bool is_regexp(VM& vm, GlobalObject& global_object, Value possible_regexp)
```

Comment:
It [looks like the `RegExp` constructor will use this as well](https://tc39.es/ecma262/#sec-regexp-pattern-flags), so let's move it to `Value.cpp`/`Value.h`, next to all the other `is_*` functions (even though those don't take a vm and GlobalObject).